test/counterParsing.cpp

```diff
@@ -142,59 +142,59 @@ TEST(ParseCounterTest, wrongType)
 TEST(ParseCounterTest, countMimeType) {
   {
     std::string counterStr = "text/html;raw=true=1";
-    CHECK(true, 1);
-    CHECK(false, 0);
-    CHECK(s.find("text/html") == 0, 1);
-    CHECK(s.find("text/html;raw=true") == 0, 1);
+    CHECK(true, 1U);
+    CHECK(false, 0U);
+    CHECK(s.find("text/html") == 0, 1U);
+    CHECK(s.find("text/html;raw=true") == 0, 1U);
```

Comment:
IMHO, the compilation warnings about signed-vs-unsigned mismatch in comparison expressions can be qualified as a compiler defect in situations where one of the expressions is a constant value within the safe range. Having said that, of course those warnings must be fixed, but in this particular case I would prefer if it was done by adjusting the definition of the `CHECK()` macro so that it uses a signed type for the variable `count`.